packages/react/src/components/Tabs/Tabs.tsx

```diff
@@ -1246,12 +1551,58 @@ export interface TabPanelsProps {
 }
 
 function TabPanels({ children }: TabPanelsProps) {
+  const prefix = usePrefix();
+  const refs = useRef<(HTMLDivElement | null)[]>([]);
+  const hiddenStates = useRef<boolean[]>([]);
+
+  useEffect(() => {
+    function handler() {
+      const tabContainer = refs.current[0]?.previousElementSibling;
+
+      // Should only apply same height to vertical Tab Panels
+      if (tabContainer?.classList.contains(`${prefix}--tabs--vertical`)) {
+        hiddenStates.current = refs.current.map((ref) => ref?.hidden || false);
+
+        // un-hide hidden Tab Panels to get heights
+        refs.current.forEach((ref) => {
+          if (ref) {
+            ref.hidden = false;
+          }
+        });
```

Comment:
If I understand the intent here, I think you could avoid having to unhide them like this by using `useLayoutEffect`. It's [specifically geared towards](https://react.dev/reference/react/useLayoutEffect#measuring-layout-before-the-browser-repaints-the-screen) logic that needs to measure the dom. 

OR maybe this justifies having a wrapper component around the tabs and the tabpanels. That way we wouldn't have to measure anything and it could be done with CSS. Honestly I'd be up for that approach over measuring with js to match heights. Are there any downsides of a wrapper element that you can think of?

Reply:
I thought the `useLayoutEffect` would work here as well! 

When trying this:
```
useLayoutEffect(() => {
      refs.current.forEach((ref) => {
        console.log(ref?.offsetHeight);
      });
});
```
<img width="1329" alt="Screenshot 2024-06-07 at 9 53 47 AM" src="https://github.com/carbon-design-system/carbon/assets/20210594/59c160a3-e860-43e7-9dc1-c51644bb9394">
^ i would only get the following printed... the only visible height 48 and the rest 0


The only side effect of a wrapper element that I can see is possibly a bit of a structure change, but there would be a lot of benefits layout wise for vertical tabs. This could also solve the issue @alisonjoseph brought up where I am using percentages.

Reply:
just confirming the above that adding a wrapper with styles such as:

```
.#{$prefix}--tabs-wrapper {
    display: flex;
  }
```

then would remove the need for using javascript and css would automatically handle the heights. If both the default and vertical tabs have this wrapper though then there would need to be some javascript to handle when to apply `display: flex` or not to handle the different layouts for vertical and default